packages/odf/components/s3-browser/cors-rules-list/CORSRulesList.tsx

```diff
@@ -0,0 +1,341 @@
+  const { t } = useCustomTranslation();
+  const tableColumns = React.useMemo<TableColumn<CORSRule>[]>(
+    () => [
+      {
+        title: t(columnInfoToTitleMap.get(tableColumnInfo[0])),
+        sort: 'ID',
+        transforms: [sortable],
+        id: tableColumnInfo[0],
+      },
+      {
+        title: t(columnInfoToTitleMap.get(tableColumnInfo[1])),
+        id: tableColumnInfo[1],
+      },
+      {
+        title: t(columnInfoToTitleMap.get(tableColumnInfo[2])),
+        id: tableColumnInfo[2],
+      },
+      {
+        title: t(columnInfoToTitleMap.get(tableColumnInfo[3])),
+        id: tableColumnInfo[3],
+      },
+      {
+        title: t(columnInfoToTitleMap.get(tableColumnInfo[4])),
+        id: tableColumnInfo[4],
+      },
+      {
+        title: t(columnInfoToTitleMap.get(tableColumnInfo[5])),
+        id: tableColumnInfo[5],
+      },
+      {
+        title: '',
+        id: tableColumnInfo[6],
+      },
+    ],
+    [t]
+  );
+
+  const [columns] = useActiveColumns({
+    columns: tableColumns,
+    showNamespaceOverride: false,
+    columnManagementID: null,
+  });
+
+  return (
+    <VirtualizedTable
+      {...props}
+      aria-label={t('Lifecycle rules list page')}
+      columns={columns}
+      Row={RuleRow}
+    />
+  );
+};
+
+const RuleRow: React.FC<RowProps<CORSRule, CustomData>> = ({
+  obj: corsRuleObj,
+  activeColumnIDs,
+  rowData,
+}) => {
+  const { t } = useCustomTranslation();
+
+  const { navigate, launcher, mutate, noobaaS3, bucketName } = rowData;
+
+  // allowed-origins
+  const ruleName = corsRuleObj?.ID;
+  const allowedOriginsLength = corsRuleObj.AllowedOrigins.length;
+  const trimmedAllowedOrigins = corsRuleObj?.AllowedOrigins.map((strOrgns) =>
+    strOrgns.trim()
+  );
+  const allowedOriginsRowValue = trimmedAllowedOrigins.includes('*')
+    ? 'All origins'
+    : allowedOriginsLength +
+      (allowedOriginsLength === 1 ? ' origin' : ' origins');
+  // allowed-methods
+  const allowedMethodsLength = corsRuleObj?.AllowedMethods.length;
+  const allowedMethodsRowValue =
+    allowedMethodsLength +
+    (allowedMethodsLength === 1 ? ' method' : ' methods');
+  // allowed-headers
+  const allowedHeadersLength = corsRuleObj?.AllowedHeaders?.length ?? 0;
+  const allowedHeadersRowValue =
+    corsRuleObj?.AllowedHeaders?.includes('*') || false
+      ? 'All headers'
+      : allowedHeadersLength + ' allowed';
+  // exposed-headers
+  const exposedHeadersLength = corsRuleObj?.ExposeHeaders?.length ?? 0;
+  const exposedHeadersRowValue = exposedHeadersLength + ' exposed';
+
+  // fallback if rule name (ID) is missing
+  const ruleHash: number = React.useMemo(() => {
+    if (!ruleName) {
+      return murmur3(JSON.stringify(deepSortObject(corsRuleObj)));
+    }
+    return null;
+  }, [ruleName, corsRuleObj]);
+
+  return (
+    <>
+      <TableData id={tableColumnInfo[0]} activeColumnIDs={activeColumnIDs}>
+        {ruleName || DASH}
```

Comment:
```suggestion
<Link to={detailsPagePath}>{ruleName || DASH}</Link>
```
const detailsPagePath = `${BUCKETS_BASE_ROUTE}/${bucketName}/permissions/cors/details?${getSearchParam()}`

Reply:
Done

Reply:
Used `<ListPageCreateLink>` method instead of `<Link>`, as we used the previous before in this code. Please let me know if this has to be changed

Reply:
`ListPageCreateLink` is for diff use... plz use `Link`...

Reply:
Done